app/internal/actor/donor.go

```diff
@@ -16,6 +16,26 @@ type Donor struct {
 	Address     place.Address
 }
 
-func (p Donor) FullName() string {
-	return fmt.Sprintf("%s %s", p.FirstNames, p.LastName)
+func (d Donor) FullName() string {
+	return fmt.Sprintf("%s %s", d.FirstNames, d.LastName)
+}
+
+func (d Donor) PossessiveFullName() string {
+	if d.LastName == "" {
+		format := "%s’s"
+
+		if d.FirstNames[len(d.FirstNames)-1:] == "s" {
+			format = "%s’"
+		}
+
+		return fmt.Sprintf(format, d.FirstNames)
+	} else {
+		format := "%s %s’s"
+
+		if d.LastName[len(d.LastName)-1:] == "s" {
```

Comment:
maybe `strings.HasSuffix(d.LastName, "s")` is clearer

Reply:
and you could cut down the duplicated logic with something like this at the start

```go
name := d.LastName
if d.LastName == "" {
  name = d.FirstNames
}
```

Reply:
Also, I wonder how this all works in Welsh... https://old.reddit.com/r/learnwelsh/comments/5oqafl/welsh_grammar_the_genitivepossessive_construction/

Perhaps for now, so we remember it doesn't actually work, pass the language in and for Cy just return "Welsh" like we do for translations?